github/codeowners.py

```diff
@@ -146,8 +145,50 @@ def resolve_email_addresses(
                 yield email_addr
             else:
                 continue
+
         else:
-            yield from resolve_team_members(
+            for member in resolve_team_members(
                 github_team_name=codeowner_entry[1:],
                 github_api=github_api,
-            )
+            ):
+                if member.email:
+                    yield member.email
+                else:
+                    logger.warning(f'no email found for GitHub user {member}')
+
+
+def resolve_gh_usernames(
```

Comment:
this duplicates "resolve_email_addresses", does it not? rather rename "resolve_email_addresses"  to "iter_codeowners", have iter_codeowners return either of email-addr, gh-username, gh-team, and have functions for filtering by type, and resolving, e.g. username -> email-addr (or none), team -> members

Reply:
nit: since this resides within the github.codeowners module, I think it is rather redundant to have the `gh`-infix in the function's name.